src/react/useInfiniteQuery.ts

```diff
@@ -19,6 +19,11 @@ export interface UseInfiniteQueryObjectConfig<TResult, TError> {
 
 // HOOK
 
+// Object syntax
```

Comment:
@tannerlinsley changed the order of overridden function types, so that correct type inference is picked for Object syntax